proctor-common/src/test/java/com/indeed/proctor/common/TestRuleEvaluator.java

```diff
@@ -85,11 +83,24 @@ public void testNonBooleanShouldFail() {
                     .hasMessageContaining("Received non-boolean return value");
         }
         {
-            // numeric result becomes String, not boolean
-            assertThatThrownBy(() -> ruleEvaluator.evaluateBooleanRule("${'tr'}${'ue'}", emptyMap()))
-                    .isInstanceOf(IllegalArgumentException.class)
-                    .hasMessageContaining("Received non-boolean return value");
+            // String 'true' is true
+            assertTrue(ruleEvaluator.evaluateBooleanRule("${'tr'}${'ue'}", emptyMap()));
```

Comment:
I do not think this is a valid format, ${'true'} would be valid but I do not think this format should be supported. cc @stevenchen-indeed for confirmation

Reply:
Yes this does not look like convention. If it does not error out, maybe it is a corner case that was not caught before,

Reply:
IMO, it's okay as I think no one writes the rule like ${'true'}. But this will help validate the custom function like author's team need: placementFilter.matchesMetadata(INDEED_PLUS)

Reply:
Looking into this we already allow it to get inputted into a test this just makes parsing it valid.

Reply:
Yeah, there was already a test with this rule (which probably should not be allowed on the UI) so I just 'fixed' it here in light of the new approach.